src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -646,6 +770,304 @@ void extendCellMapWithRemoteKeys( VTKMeshGenerator::CellMapType & cellMap )
+{
+  std::vector< localIndex > nodeOrder( 4 );
+
+// Check if the orientation is correct
+  real64 vectorA[3];
+  real64 vectorB[3];
+  real64 vectorC[3];
+  real64 vectorN[3];
+
+  for( localIndex i = 0; i < 3; ++i )
+  {
+    vectorA[i] = cell->GetPoints()->GetPoint( 1 )[i] - cell->GetPoints()->GetPoint( 0 )[i];
+    vectorB[i] = cell->GetPoints()->GetPoint( 2 )[i] - cell->GetPoints()->GetPoint( 0 )[i];
+    vectorC[i] = cell->GetPoints()->GetPoint( 3 )[i] - cell->GetPoints()->GetPoint( 0 )[i];
+  }
+
+  LvArray::tensorOps::crossProduct( vectorN, vectorA, vectorB );
+
+  if( LvArray::tensorOps::AiBi< 3 >( vectorN, vectorC ) > 0 ) // The orientation is correct
+  {
+    nodeOrder[0] = 0;
+    nodeOrder[1] = 1;
+    nodeOrder[2] = 2;
+    nodeOrder[3] = 3;
+  }
+  else // The orientation is incorrect, renumber nodes
+  {
+    nodeOrder[0] = 0;
+    nodeOrder[1] = 2;
+    nodeOrder[2] = 1;
+    nodeOrder[3] = 3;
+  }
+
+  return nodeOrder;
+}
+/**
+ * @brief Get the hexahedron node ordering from a vtk polyhedron (42)
+ *
+ * @param cell The VTK cell, type polyhedron (42)
+ * @return std::vector< localIndex >  list of nodes
+ */
+std::vector< localIndex > getHexahedronNodeOrderingFromPolyhedron( vtkCell * const cell )
+{
+  localIndex iFace;
+  std::vector< localIndex > nodeOrder( 8 );
+  std::map< localIndex, localIndex > G2L;
+
+// Generate global to local map
+  for( localIndex iPoint = 0; iPoint < 8; ++iPoint )
+    G2L.insert( {cell->GetPointId( iPoint ), iPoint} );
+
+// Assuming the input parameters are correct, take the first quad
+  iFace = 0;
+
+// Get global pointIds for the first quad
+  nodeOrder[0] = cell->GetFace( iFace )->GetPointId( 0 );
+  nodeOrder[1] = cell->GetFace( iFace )->GetPointId( 1 );
+  nodeOrder[2] = cell->GetFace( iFace )->GetPointId( 3 );
+  nodeOrder[3] = cell->GetFace( iFace )->GetPointId( 2 );
+
+// Using the edges, generate the global pointIds for the opposit quad
+  for( localIndex iEdge = 0; iEdge < 12; ++iEdge )
+  {
+    auto edgeNode0 = cell->GetEdge( iEdge )->GetPointId( 0 );
+    auto edgeNode1 = cell->GetEdge( iEdge )->GetPointId( 1 );
+    auto it0 = std::find( &nodeOrder[0], &nodeOrder[4], edgeNode0 );
+    auto it1 = std::find( &nodeOrder[0], &nodeOrder[4], edgeNode1 );
+
+    if( it0 != &nodeOrder[4] && it1 == &nodeOrder[4] )
+      nodeOrder[ 4 + std::distance( &nodeOrder[0], it0 ) ] = edgeNode1;
+
+    if( it0 == &nodeOrder[4] && it1 != &nodeOrder[4] )
+      nodeOrder[ 4 + std::distance( &nodeOrder[0], it1 ) ] = edgeNode0;
+  }
+
+// Convert global numbering to local numbering
+  for( localIndex iPoint = 0; iPoint < 8; ++iPoint )
+    nodeOrder[iPoint] = G2L.find( nodeOrder[iPoint] )->second;
+
+  real64 Xlocal[8][3];
+  real64 cellVolume;
+  for( localIndex iPoint = 0; iPoint < 8; ++iPoint )
+    for( localIndex i = 0; i < 3; ++i )
+      Xlocal[iPoint][i] = cell->GetPoints()->GetPoint( nodeOrder[iPoint] )[i];
```

Comment:
Extract `cell->GetPoints()` and extract `cell->GetPoints()->GetPoint( nodeOrder[iPoint] )` too.